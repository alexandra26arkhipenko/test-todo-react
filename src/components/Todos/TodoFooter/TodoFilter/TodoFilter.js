import {
  TODO_FILTER_ACTIVE_BTN,
  TODO_FILTER_COMPLETED_BTN,
  TODO_FOOTER_FILTER_ALL,
} from "../../../../constants/constants";
import Button from "../../../UI/Button/Button";

const TodoFilter = (props) => {
  const clickHandler = (event) => {
    props.onFilter(event.target.value);
  };

  return (
    <div>
      <Button
        type="button"
        onClick={clickHandler}
        value={TODO_FOOTER_FILTER_ALL}
      >
        {TODO_FOOTER_FILTER_ALL}
      </Button>
      <Button
        type="button"
        onClick={clickHandler}
        value={TODO_FILTER_ACTIVE_BTN}
      >
        {TODO_FILTER_ACTIVE_BTN}
      </Button>
      <Button
        type="button"
        onClick={clickHandler}
        value={TODO_FILTER_COMPLETED_BTN}
      >
        {TODO_FILTER_COMPLETED_BTN}
      </Button>
    </div>
  );
};

export default TodoFilter;
