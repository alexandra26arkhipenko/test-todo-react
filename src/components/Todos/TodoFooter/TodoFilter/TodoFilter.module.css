.filter button:hover {
    color: rgb(114, 12, 12);
}