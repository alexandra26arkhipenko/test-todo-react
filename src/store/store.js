import { configureStore } from "@reduxjs/toolkit";
import { localStorageMiddleware } from "./middleware/localStorageMiddleware";

import todoReducer from "./todo";

const store = configureStore({
  reducer: todoReducer,
  middleware: [localStorageMiddleware],
});

export default store;
