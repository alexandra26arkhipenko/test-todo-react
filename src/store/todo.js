import { createSlice } from "@reduxjs/toolkit";
import { v4 as uuid } from "uuid";
import { getItemFromLocalStorage } from "../repository/localStorageRepository";
import { TODO_FOOTER_FILTER_ALL } from "../constants/constants";

const savedTodos = getItemFromLocalStorage("todos");
let todos = [];

if (savedTodos) {
  todos = JSON.parse(savedTodos);
}

const initialState = {
  todos: todos,
  filter: "",
};

const todoSlice = createSlice({
  name: "counter",
  initialState,
  reducers: {
    addTodo(state, action) {
      const newTodo = {
        id: uuid(),
        text: action.payload,
        isCompleted: false,
      };

      state.todos.unshift(newTodo);
    },
    compliteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos[index].isCompleted = !state.todos[index].isCompleted;
      }
    },
    deleteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos.splice(index, 1);
      }
    },
    deleteAllCompletedTodos(state) {
      state.todos = state.todos.filter((todo) => !todo.isCompleted);
    },
    filterTodos(state, action) {
      state.filter = action.payload;
    },
    reorderTodos(state, action) {
      let { dragIndex, dropIndex, filteredTodos } = action.payload;

      if (state.filterValue !== TODO_FOOTER_FILTER_ALL) {
        dragIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dragIndex].id
        );
        dropIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dropIndex].id
        );
      }

      const updatedTodos = [...state.todos];
      const [reorderedItem] = updatedTodos.splice(dragIndex, 1);
      updatedTodos.splice(dropIndex, 0, reorderedItem);
      state.todos = updatedTodos;
    },
  },
});

export const todoActions = todoSlice.actions;
export default todoSlice.reducer;
