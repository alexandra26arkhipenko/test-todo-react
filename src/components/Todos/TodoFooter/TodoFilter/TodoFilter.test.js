import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../../store/store";
import TodoFilter from "./TodoFilter";

const props = {
  onFilter: jest.fn(),
};

describe("TodoFilter", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <TodoFilter {...props} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
