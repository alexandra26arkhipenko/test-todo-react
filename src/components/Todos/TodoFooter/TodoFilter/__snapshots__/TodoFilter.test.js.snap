// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilter should render correctly 1`] = `
<div>
  <div>
    <button
      class="button undefined"
      type="button"
      value="All"
    >
      All
    </button>
    <button
      class="button undefined"
      type="button"
      value="Active"
    >
      Active
    </button>
    <button
      class="button undefined"
      type="button"
      value="Completed"
    >
      Completed
    </button>
  </div>
</div>
`;
